### PR TITLE
Adds 2 new examples

### PR DESCRIPTION
**Pull request by nayabur**
Hi! I created two new demos for the PyshGP examples folder. One is called triple.py, and it's a simple example that walks users through a GP run. The other is called rectangle_area.py, and it demonstrates how to create custom types and instructions.

### CODE REVIEW

File: examples/rectangle_area.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        self.height = height
+        self.width = width
+
+    # checks the equality of two Rectangle objects
+    def __eq__(self, other):
+        if isinstance(other, Rectangle):
+            return self.height == other.height and self.width == other.width
+        return False
+
+    # returns the Rectangle object's representation
```
Comment (erp12) [inline examples/rectangle_area.py]: The comments before each method don't tell us much that we don't get from the name of the method. I think the code would be slightly cleaner without them.

File: examples/rectangle_area.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+"""
+Next, we create a type library that specifies we will be synthesizing programs that manipulate 
+    "floats" (built-in to pyshgp) and "rectangles" (custom for this problem).
+"""
+
+type_library = (
+    PushTypeLibrary(register_core=False)
+        .register(PushFloat)
+        .create_and_register(name="rectangle", underlying_types=(Rectangle,), coercion_func=to_rect)
+)
```
Comment (erp12) [inline examples/rectangle_area.py]: Very minor and nit-picky suggestion: Although many text editors get this wrong (including the one I use), the python style guides suggest vertical alignment when using`()` for a multi-line statement. For example:

```py
type_library = (
    PushTypeLibrary(register_core=False)
    .register(PushFloat)
    .create_and_register(name="rectangle", underlying_types=(Rectangle,), coercion_func=to_rect)
)
```

File: examples/rectangle_area.py [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+The instruction set will register all core instructions that can be supported 
+    using only exec, code, float, and rectangle types because the only core PushType we registered was "PushInt"
+For example, the instruction int_from_float will NOT be registered because
+    our type library does not define a type that would support the "int" stack.
+"""
+
+instruction_set = (
+    InstructionSet(type_library=type_library, register_core=True)
+        .register(rectangle_areas_instruction)
+        .register(rectangle_from_floats_instruction)
```
Comment (erp12) [inline examples/rectangle_area.py]: Same whitespace style comment as above.

**Review by erp12 — CHANGES_REQUESTED** [review]
These examples are great, thanks for putting them together!  I like that we another demonstration of adding custom types and instructions. I have a few minor requests before we merge this pull request.

1. Let's make sure that we run these examples before every public release to check for any changes to pyshgp that may have broke the your code. To help with this, we use a deployment automation tool called [tox](https://tox.readthedocs.io/en/latest/). The only change we need to make is in the `tox.ini` file in the root the project. Just add a couple lines to this section that runs all the example files. You can use the other lines as an template for the syntax.

https://github.com/erp12/pyshgp/blob/6ff3f8e1e3491bea7529687cff2411256db406ad/tox.ini#L37-L47

2. It looks like the start of the commit message is "Add files via upload". That seems like an auto-generated message to me. Git only shows the first line of the commit message in the history, so I think it is helpful to have a 1 sentence summary of what the commit changes. Something like "Adds 2 new examples". Can we change that?

Other than those small changes, this PR is ready to be merged! I think these examples will serve future new users well.

**Comment by nayabur** [comment]
I've made all the changes requested. Thank you!

**Comment by erp12** [comment]
Looks great, and merged! Thanks for the contribution :)

**Pull request merged**